### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by DianaVashti**

### CODE REVIEW

File: controllers/admins_controller.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    const adminProps = _.pick(req.body, ['email', 'password']);
+    const admin = new Admin(adminProps);
+
+    admin.save().then(() => { return admin.generateAuthToken() })
+      .then((token) => {
+        res.header('x-auth', token).send(admin)
+      })
+      .catch(next);
+  }
+};
```
Comment (DianaVashti) [inline controllers/admins_controller.js]: 👍

File: controllers/helper_functions.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      query.city = criteria.city;
+    };
+
+    if (criteria.type) {
+      query.type = criteria.type;
+    };
+
+    return query;
+  }
+}
```
Comment (DianaVashti) [inline controllers/helper_functions.js:19]: V-cool.

File: controllers/reports_controller.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  },
+
+  searchEditedReports(req, res, next) {
+    const criteria = req.query;
+    
+    Report.find(helpers.buildQuery(criteria))
+      .then((reports) => res.send(reports))
+      .catch(next);
+  }
+};
```
Comment (DianaVashti) [inline controllers/reports_controller.js:59]: 👍

File: models/admin.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+const AdminSchema = new Schema({
+  email: {
+    type: String,
+    required: true,
+    trim: true,
+    unique: true,
+    minlength: 3,
+    validate: {
+      validator: validator.isEmail,
```
Comment (DianaVashti) [inline models/admin.js:16]: Mongoose is so cool, excited there's an isEmail validation built-in!

File: models/perp_schema.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+		required: true
+	},
+	attributes: {
+		type: String,
+		required: true
+	},
+	vehicle: String
+});
+
+module.exports = PerpSchema;
```
Comment (DianaVashti) [inline models/perp_schema.js:43]: I can't remember, did we decide we wanted all these fields to be required? I thought we wanted them to exist but not be required?

File: models/service.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    type: String,
+    required: true
+  }
+});
+
+ServiceSchema.index({ 'name': 'text' });
+
+const Service = mongoose.model('service', ServiceSchema);
+
+module.exports = Service;
```
Comment (DianaVashti) [inline models/service.js:39]: I also can't remember if our resources page was going to be a link to one that SJI already provides? Is that a thing that exists?

File: routes/routes.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  // All can access a single services
+  app.get('/api/services/:id', ServicesController.getOneService);
+
+  // Admin can update an existing service
+  app.put('/api/services/:id', ServicesController.updateService);
+
+  // Admin can delete a service
+  app.delete('/api/services/:id', ServicesController.removeService);
+
+};
```
Comment (DianaVashti) [inline routes/routes.js]: 👍

File: test/app_test.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+describe('The express app', () => {
+	it('handles a GET request to /api', (done) => {
+		request(app)
+			.get('/api')
+			.end((err, res) => {
+				assert(res.body.hello === 'world');
+				done();
+			});
+	});
+});
```
Comment (DianaVashti) [inline test/app_test.js:14]: Was so much indentation intended here? Looks weird to me, but I am not necessarily opposed.

File: test/controllers/dummy_data.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    zipCode: '94111',
+    phone: '4155583333',
+    type: 'legal'
+  },
+
+  adminUserOne: {
+    email: 'test@test.com',
+    password: '123456'
+  }
+};
```
Comment (DianaVashti) [inline test/controllers/dummy_data.js]: 👍

**Review by DianaVashti — COMMENTED** [review]
Looks great. Reading this made me really appreciate mongo/mongoose. Looking forward to hooking up the front end tomorrow 👍

File: test/app_test.js (hunk 8 above)
Comment (spencerdezartsmith) [reply test/app_test.js:14]: OMG no! Thats so ugly. I will lint this to make sure everything is good.

File: models/perp_schema.js (hunk 5 above)
Comment (spencerdezartsmith) [reply models/perp_schema.js:43]: They were required on the current form thats why I changed it.

**Pull request merged**